classes/battle/Fighter.php

```diff
@@ -338,10 +349,22 @@ public function calcDamage(Jutsu $attack, bool $disable_randomness = false): flo
             echo "Damage/boost/nerf: $damage / {$off_boost} / {$off_nerf}} <br />";
         }
 
-        $damage = round(
-          ($damage * (1 + $off_boost)) - $off_nerf,
-          2
-        );
+        $off_modifier = 1 + $off_boost - $off_nerf;
+        // if net offense nerf, apply caps
+        if ($off_modifier < 1) {
+            $nerf_percent = 1 - $off_modifier;
+            // if higher than soft cap, apply penalty
+            if ($nerf_percent > BattleManager::OFFENSE_NERF_SOFT_CAP) {
+                $nerf_percent = (($nerf_percent - BattleManager::OFFENSE_NERF_SOFT_CAP) * BattleManager::OFFENSE_NERF_SOFT_CAP_RATIO) + BattleManager::OFFENSE_NERF_SOFT_CAP;
```

Comment:
```suggestion
                $nerf_percent = (
                    ($nerf_percent - BattleManager::OFFENSE_NERF_SOFT_CAP) *    BattleManager::OFFENSE_NERF_SOFT_CAP_RATIO
                ) + BattleManager::OFFENSE_NERF_SOFT_CAP;
```
breaking this into multiple lines makes it slightly more readable I think